src/physics/model.h

```diff
@@ -38,14 +39,67 @@ class ModelBase
     	const std::array<dealii::Tensor<1,dim,real>,nstate> &solution_gradient,
         const dealii::types::global_dof_index cell_index) const = 0;
 
-    /// Source terms additional to the baseline physics
+    /// Convective Numerical Split Flux for split form additional to the baseline physics
+    virtual std::array<dealii::Tensor<1,dim,real>,nstate> 
+    convective_numerical_split_flux (
+        const std::array<real,nstate> &soln_const, 
+        const std::array<real,nstate> &soln_loop) const = 0;
+
+    /// Spectral radius of convective term Jacobian.
+    /** Used for scalar dissipation
+     */
+    virtual std::array<real,nstate> convective_eigenvalues (
+        const std::array<real,nstate> &/*solution*/,
+        const dealii::Tensor<1,dim,real> &/*normal*/) const = 0;
+
+    /// Maximum convective eigenvalue used in Lax-Friedrichs
+    virtual real max_convective_eigenvalue (const std::array<real,nstate> &soln) const = 0;
+
+    /// Physical source terms 
```

Comment:
```suggestion
    /// Physical source terms  additional to the baseline physics
```

Reply:
fixed.
I add more comments which explained more about the additional convective, dissipative and physical source terms from ModelBase.